src/components/Text/src/Text.vue

```diff
@@ -123,9 +132,11 @@ export default {
 		},
 		inlineStyles() {
 			const { fonts } = this.theme;
+			const resolveProps = this.resolveVariant();
+
 			return {
-				fontFamily: this.resolvedFontFamily,
-				fontWeight: this.resolvedFontWeight,
+				fontFamily: resolveProps ? this.resolvedFontFamily : this.fontFamily,
+				fontWeight: resolveProps ? this.resolvedFontWeight : this.fontWeight,
```

Comment:
This code is very confusing. If `this.fontFamily` is ever anything other than `undefined` then that's what `this.resolvedFontFamily` will be set to, same for the font weights. The only time they are different are if `this.fontFamily` is `undefined` in which case `this.resolvedFontFamily` will get the value `inherit` from the default theme so why not just check that? Or why not just check for the value `inherit` instead of using this convoluted logic? Also why is the logic depend on whether or not it's a `paragraph`/`title` variant vs a `headline`/`label` variant?